sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1661,6 +1661,8 @@ class GpuCachedDoublePassWindowIterator(
 
   override def isRunningBatched: Boolean = true
 
+  private var firstPassIter: Option[Iterator[(Array[cudf.ColumnVector], ColumnarBatch)]] = None
```

Comment:
Could we get an explanation of what is happening with this iterator?.  Why is there a pair of what appears to be two tables of data?